### PR TITLE
Use return codes in multiple shooting

### PR DESCRIPTION
**Pull request by adrhill**
Previously, failure of integration of one of the groups in multiple shooting (e.g. due to `:DtLessThanMin`) lead to an interruption of training as there would be a dimension mismatch between the training data and the prediction.
 
## Changes
1. In this PR,  infinite loss is returned on failure of integration by using the solution return codes. 
    I wasn't sure how to add tests for this, so let me know if you have an idea.

1. I also included `test/multiple_shoot.jl` in `test/runtest.jl`, as the tests previously weren't run in CI. I wasn't sure which section to include them in, so you might want to change this.

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]
yes good idea.

**Comment by ChrisRackauckas** [comment]
I'm not sure how to test it either. Maybe do LV with some weirdly bad starting parameters?

**Comment by adrhill** [comment]
I added a test setting the solver options to `maxiters=1`, which seems to do the job.
Is there any way to suppress the expected/intended warning? Or should I just leave it like this?

**Comment by ChrisRackauckas** [comment]
The user can pass the `verbose=false` kwarg to remove it.
